test/e2e/app-dir/ppr-full/ppr-full.test.ts

```diff
@@ -453,20 +453,41 @@ describe('ppr-full', () => {
 
           // Now let's revalidate the page.
           await next.fetch(
-            '/api/revalidate?pathname=/fallback/dynamic/params/[slug]'
+            `/api/revalidate?pathname=${encodeURIComponent(pathname)}`
           )
 
-          // We expect to get a revalidated shell, not the same one as before.
+          // We expect to get the fallback shell.
           $ = await next.render$(pathname)
-          expect($('[data-layout]').data('layout')).not.toBe(fallbackID)
+          expect($('[data-layout]').data('layout')).toBe(fallbackID)
 
           // Let's wait for the page to be revalidated.
+          let revalidatedDynamicID: string
           await retry(async () => {
             $ = await next.render$(pathname)
-            expect($('[data-layout]').data('layout')).not.toBe(dynamicID)
+            revalidatedDynamicID = $('[data-layout]').data('layout') as string
+            expect(revalidatedDynamicID).not.toBe(dynamicID)
+            expect(revalidatedDynamicID).not.toBe(fallbackID)
           })
         })
       })
+
+      it('should support client layouts', async () => {
+        const $ = await next.render$('/fallback/client/params/slug-01')
+
+        let selector = $(
+          '[data-file="app/fallback/client/params/[slug]/loading"]'
+        )
+        expect(selector.length).toBe(1)
+        expect(selector.closest('[hidden]').length).toBe(0)
+
+        selector = $('[data-file="app/fallback/client/params/[slug]/layout"]')
+        expect(selector.length).toBe(1)
+        expect(selector.closest('[hidden]').length).toBe(0)
+
+        selector = $('[data-file="app/fallback/client/params/[slug]/page"]')
+        expect(selector.length).toBe(1)
+        expect(selector.closest('[hidden]').length).toBe(1)
```

Comment:
The test is not actually consuming the params passed to the layout. The loading boundary doesn't come into play because we need to suspend in the layout which is above the loading boundary